reference/nonfungible-token.scilla

```diff
@@ -85,125 +87,171 @@ transition balanceOf(address: ByStr20)
   event e
 end
 
-(* @dev:    Mint new tokens. Only contractOwner can mint the token *)
-(* @param:  to      - Address of the token recipient               *)
-(* @param:  tokenId - ID of the new token minted                   *)
-(* Returns error message code_token_exist if token exists          *)
-transition transferSingle(to: ByStr20, tokenId: Uint256)
-
-  (* Sender must be the contract owner *)
-  isAuthorized = builtin eq contractOwner _sender;
-  match isAuthorized with
+(* @dev:    Mint new tokens. Only contractOwner can mint. *)
+(* @param:  to      - Address of the token recipient      *)
+(* @param:  tokenId - ID of the new token minted          *)
+(* Returns error message CodeTokenExists if token exists  *)
+transition mint(to: ByStr20, tokenId: Uint256)
+  (* Check if token exists *)
+  tokenExist <- exists tokenOwners[tokenId];
+  match tokenExist with
   | True =>
-    (* Check if token exists *)
-    tokenExist <- exists tokenOwnerMap[tokenId];
-    match tokenExist with
+    (* Token exists, return error code *)
+    err = CodeTokenExists;
+    MakeError err
+  | False =>
+    (* Prevent transferring of NFT to the current contract address *)
+    isContractAddress = builtin eq to _this_address;
+    match isContractAddress with
+    | False =>
+      (* Check if sender is contractOwner *)
+      isContractOwner = builtin eq _sender contractOwner;
+      match isContractOwner with
+      | True =>
+        (* Mint new token *)
+        tokenOwners[tokenId] := to;
+        (* Add to owner's count *)
+        userCount <- ownedTokenCount[to];
+        match userCount with
+        | Some val =>
+          (* Append to existing results *)
+          val = builtin add val one;
```

Comment:
I see at least two occurrences of adding one to the token count and subtracting one to it. Maybe these two operations can be abstracted into two library functions that take an `Option {Uint32}` and return `Uint32` by doing the operation.